Sources/Commands/SwiftTool.swift

```diff
@@ -573,61 +573,22 @@ public class SwiftTool {
     }
 
     func getAuthorizationProvider() throws -> AuthorizationProvider? {
-        var providers = [AuthorizationProvider]()
-
-        // netrc file has higher specificity than keychain so use it first
-        try providers.append(contentsOf: self.getNetrcAuthorizationProviders())
-
-#if canImport(Security)
-        if self.options.keychain {
-            providers.append(KeychainAuthorizationProvider(observabilityScope: self.observabilityScope))
-        }
-#endif
-
-        return providers.isEmpty ? .none : CompositeAuthorizationProvider(providers, observabilityScope: self.observabilityScope)
-    }
-
-    func getNetrcAuthorizationProviders() throws -> [NetrcAuthorizationProvider] {
-        guard options.netrc else {
-            return []
-        }
-
-        var providers = [NetrcAuthorizationProvider]()
-
-        // Use custom .netrc file if specified, otherwise look for it within workspace and user's home directory.
-        if let configuredPath = options.netrcFilePath {
-            guard localFileSystem.exists(configuredPath) else {
-                throw StringError("Did not find .netrc file at \(configuredPath).")
-            }
-
-            providers.append(try NetrcAuthorizationProvider(path: configuredPath, fileSystem: localFileSystem))
+        var configuration = Workspace.Authorization.Configuration()
+        if !options.netrc {
+            configuration.netrc = .disabled
+        } else if let configuredPath = options.netrcFilePath {
+            configuration.netrc = .custom(configuredPath)
         } else {
-            // User didn't tell us to use these .netrc files so be more lenient with errors
-            func loadNetrcNoThrows(at path: AbsolutePath) -> NetrcAuthorizationProvider? {
-                guard localFileSystem.exists(path) else { return nil }
-
-                do {
-                    return try NetrcAuthorizationProvider(path: path, fileSystem: localFileSystem)
-                } catch {
-                    self.observabilityScope.emit(warning: "Failed to load .netrc file at \(path). Error: \(error)")
-                    return nil
-                }
-            }
-
-            // Workspace's .netrc file should be consulted before user-global file
-            // TODO: replace multiroot-data-file with explicit overrides
-            if let localPath = try? (options.multirootPackageDataFile ?? self.getPackageRoot()).appending(component: ".netrc"),
-               let localProvider = loadNetrcNoThrows(at: localPath) {
-                providers.append(localProvider)
-            }
-
-            let userHomePath = localFileSystem.homeDirectory.appending(component: ".netrc")
-            if let userHomeProvider = loadNetrcNoThrows(at: userHomePath) {
-                providers.append(userHomeProvider)
-            }
+            let rootPath = try options.multirootPackageDataFile ?? self.getPackageRoot()
```

Comment:
I know this is just moving the code, but I am not sure if it this is actually correct for multi root. This will be a file path, not a directory, so essentially never work.

Reply:
interesting, the entire code in SwiftTool treats `multirootPackageDataFile` as the root directory, sounds like that's working (and that we are missing some tests?)

Reply:
Oh, I see how it works, `options.multirootPackageDataFile` is actually pointing to a `.xcworkspace` which is a directory. I think we should change this to use `xcshareddata` because this is a configuration file.

Reply:
To be clear, what I mean is use `multiRootPackageDataFile.appending(components: "xcshareddata")` as the root path here.

Reply:
sounds like good material for follow on PR? it is used in various spots